app/services/project_service.rb

```diff
@@ -14,28 +14,39 @@ class ProjectService
     attr_accessor :project_data_source
 
     def initialize(project_data_source: nil)
-      unless project_data_source.nil?
-        raise "project_data_source must be descendant of #{ProjectDataSource.name}" unless project_data_source.class <= ProjectDataSource
+      if !project_data_source.nil? && project_data_source.class > ProjectDataSource
```

Comment:
No, I changed it because I wasn't a big fan of having an `unless` guard within an `unless` block. I thought it would be better to have one boolean expression, so it was easier to reason about.